gollm/ollama.go

```diff
@@ -58,10 +59,20 @@ var _ Client = &OllamaClient{}
 func NewOllamaClient(ctx context.Context, opts ClientOptions) (*OllamaClient, error) {
 	// Create custom HTTP client with SSL verification option from client options
 	httpClient := createCustomHTTPClient(opts.SkipVerifySSL)
-	client := api.NewClient(envconfig.Host(), httpClient)
 
+	ollamaURL := envconfig.Host()
+	if opts.URL != nil && opts.URL.Host != "" {
+		ollamaURL.Host = opts.URL.Host
+	}
+	client := api.NewClient(ollamaURL, httpClient)
+
+	defaultModel := defaultOllamaModel
+	if opts.URL != nil {
+		defaultModel = opts.URL.Query().Get("model")
```

Comment:
hmm...we should only override the defaultModel if the url has `model` query parameter.. .right ? or am I missing something.